tests/testthat/test-examples.R

```diff
@@ -0,0 +1,127 @@
+# this test requires a `man` directory in the `tests/testthat` directory
+# (presumably symlinked to the package root `man` directory to avoid duplication)
+# this also requires `devtools::document()` to be run before running the tests
+
+rd_files <- function() {
+  list.files(testthat::test_path("man"), pattern = "\\.[Rr]d$", full.names = TRUE)
```

Comment:
It seems to work (locally) without `man` folder on the testthat directory.

I've run the test file until the first `if (grepl("tm_g", i)) break` enters (3 example apps are run)

Am I missing something?

```suggestion
  list.files(
    normalizePath(file.path(testthat::test_path(), "..", "..", "man")),
    pattern = "\\.[Rr]d$",
    full.names = TRUE
  )
```

Reply:
It's just because I think that jumping outside of `test/` directory inside tests not very elegant. I have seen somewhere that it's not so safe to do this - but I cannot recall all the details. It was something about many ways of executing tests - `devtools::test()`, `R CMD CHECK` etc. but these all are ok so I'm not sure about this tbh

I did break this rule already in `load_all()` step but this is just a temporary workaround that I hope will be removed at some point. This step should remain unchanged.

Because I cannot confirm my suspicions I'm inclined to accept your suggestion (and also rm symlink). It will be easier to copy&paste to other repos. WDYT?

Reply:
I lean towards removal for all the reasons you mentioned and I don't like having a `symlink` to documentation on the `testthat` folder.
- My brain immediately wonders how it behaves in Windows/MacOSX, or that creates the opportunity for a weird bug-hunting mission if it's accidentally removed by someone.

The safe argument is strong though, we could create a symlink during testing that's removed on exit.

Reply:
ok let me remove it then
if someone ever find issues then it won't be super difficult to re-introduce it again.

Reply:
Commited this and we got an error in CI:)
```
  Error in UseMethod("xml_add_child") : 
    no applicable method for 'xml_add_child' applied to an object of class "NULL"
  Calls: test_check ... o_apply -> lapply -> FUN -> <Anonymous> -> <Anonymous>
  Execution halted
```

This means that there is a warning in the test code outside of test_that (it's coming from JunitReporter and I already reported this to testthat folks)
I have changed this slightly into `normalizePath(..., mustWork = TRUE)` to throw error instead of warning and this is what I got locally from rcmdcheck:

```
══ Failed tests ════════════════════════════════════════════════════════════════
── Error ('test-examples.R:6:3'): (code run outside of `test_that()`) ──────────
Error in `normalizePath(file.path(testthat::test_path(), "..", "..", "man"), 
    mustWork = TRUE)`: path[1]="./../../man": No such file or directory
Backtrace:
    ▆
 1. └─teal.modules.clinical (local) rd_files() at test-examples.R:110:1
 2.   ├─base::list.files(...) at test-examples.R:6:3
 3.   └─base::normalizePath(...)
 ```

Looks like I can now confirm my concerns. Do you want to give a shot on this or we come back to symlinks?

Reply:
One last try, how about something like this?

```r
rd_files <- function() {
  man_directory <- if (grepl(".Rcheck", getwd())) {
    gsub("tests/testthat", "00_pkg_src/teal.modules.clinical/man", getwd())
  } else {
    file.path(testthat::test_path(), "..", "..", "man")
  }

  list.files(
    normalizePath(man_directory, mustWork = TRUE),
    pattern = "\\.[Rr]d$",
    full.names = TRUE
  )
}
```

With this, the tests passed using `rcmdcheck::rcmdcheck()`, `devtools::test()`, and `devtools::check()` locally.
I'm just unsure how this would behave with CI/CD. 

PS: I did see other error unrelated with tests. Do you see this too?
```r
 checking running R code from vignettes ...
    ‘generate_tmc_test_data.Rmd’ using ‘UTF-8’... OK
    ‘quickstart_substitute.Rmd’ using ‘UTF-8’... failed
    ‘teal-modules-clinical.Rmd’ using ‘UTF-8’... OK
   ERROR
  Errors in running code in vignettes:
  when running code in ‘quickstart_substitute.Rmd’
    ...
  +     b)
  
  > non_evaluated_expression
  a + b
  
  > eval(non_evaluated_expression)
  
    When sourcing ‘quickstart_substitute.R’:
  Error: object 'a' not found
  Execution halted
```

Reply:
Thanks Dony. I think that using `getwd()` is even less secure but I enhanced your snipped a little and pushed a new version that should be good. Please do check it out.